content/vault/v1.21.x/content/docs/deploy/run-as-zcx-cluster.mdx

```diff
@@ -0,0 +1,317 @@
+  ```
+
+1. Verify image integrity.
+
+  ```shell-session
+  $ docker images | grep vault-enterprise
+  ```
+
+## Step 2: Persistent volume creation
+
+Create Docker volumes for configuration and data persistence on each node.
+
+1. Create volume for Vault data storage (Raft backend, audit logs).
+
+  ```shell-session
+  $ docker volume create vault-data
+  ```
+
+1. Create volume for Vault configuration files.
+
+  ```shell-session
+  $ docker volume create vault-config
+  ```
+
+1. Verify volume creation.
+
+  ```shell-session
+  $ docker volume ls | grep vault
+  ```
+
+Volume mount points:
+
+- **`vault-data`**: `/vault/data` (stores Raft snapshots, encrypted storage backend)
+- **`vault-config`**: `/vault/config.d` (HCL configuration files)
+
+## Step 4: Vault configuration file setup
+
+Create Vault configuration on each node with node-specific parameters.
+
+1. Access config volume.
+
+  ```shell-session
+  $ docker run --rm -it -v vault-config:/vault alpine sh
+  ```
+
+1. Create configuration file for Leader and follower vault.
+
+  ```hcl
+  ui = true
+  disable_mlock = true
+
+  # --- Listener Configuration ---
+  listener "tcp" {
+    address              = "0.0.0.0:8200"
+    tls_disable          = 0
+    tls_cert_file        = "/vault/vault.pem"
+    tls_key_file         = "/vault/vault.key"
+    tls_client_ca_file   = "/vault/ca.pem"
+  }
+
+  # --- Raft Storage (Leader Node) ---
+  storage "raft" {
+    path    = "/vault/data"
+    node_id = "<LEADER_NODE_ID>"  # e.g., "node1"
+
+    # No retry_join block needed for leader
+  }
+
+  # --- Cluster Networking ---
+  api_addr     = "https://<LEADER_IP>:8200"
+  cluster_addr = "https://<LEADER_IP>:8201"
+  ```
+
+  Exposes Vault on port 8200 for API and UI traffic.
+
+   - **`tls_disable`** = 0 → TLS is enabled (mandatory for secure cluster communication).
+
+### Certificates:
+
+- **`vault.pem`** → TLS certificate
+
+- **`vault.key`** → Private key
+
+- **`ca.pem`** → CA used to validate mutual TLS between nodes
+
+This ensures encrypted traffic between Vault clients, HAProxy, and other nodes.
+
+Similarly create configuration for follower Vault nodes.
+
+```hcl
+ui = true
+disable_mlock = true
+
+# --- Listener Configuration ---
+listener "tcp" {
+  address            = "0.0.0.0:8200"
+  tls_disable        = 0
+  tls_cert_file      = "/vault/vault.pem"
+  tls_key_file       = "/vault/vault.key"
+  tls_client_ca_file = "/vault/ca.pem"
+}
+
+# --- Raft Storage (Follower Node) ---
+storage "raft" {
+  path    = "/vault/data"
+  node_id = "<FOLLOWER_NODE_ID>"   # e.g., "node2" or "node3"
+
+  retry_join {
+    leader_api_addr      = "https://<LEADER_IP>:8200"
+    leader_ca_cert_file  = "/vault/ca.pem"
+  }
+}
+
+# --- Cluster Networking ---
+api_addr     = "https://<FOLLOWER_IP>:8200"
+cluster_addr = "https://<FOLLOWER_IP>:8201"
+```
+
+## Step 5: Vault container deployment
+
+Deploy Vault container on each zCX node.
+
+1. Set your Vault Enterprise license (export before running).
```

Comment:
```suggestion
1. Set and export the Vault license environment variable, `VAULT_LICENSE` to
   your Enterprise license token.
```

Where are you setting this environment variable? In which terminal? Would it be easier for folks to use the configuration parameter and add a license file to the config mount instead?